Applications/LogisticRegression/jni/main.cpp

```diff
@@ -161,64 +160,87 @@ int main(int argc, char *argv[]) {
 
   srand(time(NULL));
 
-  auto data_train = ml::train::createDataset(ml::train::DatasetType::GENERATOR,
-                                             getSample_train);
-
   /**
    * @brief     Create NN
    */
-  std::vector<std::vector<float>> inputVector, outputVector;
-  nntrainer::NeuralNetwork NN;
+  std::unique_ptr<ml::train::Model> model;
+  model = ml::train::createModel(ml::train::ModelType::NEURAL_NET);
 
   /**
    * @brief     Initialize NN with configuration file path
    */
 
   try {
-    NN.loadFromConfig(config);
-    NN.compile();
-    NN.initialize();
+    model->load(config, ml::train::ModelFormat::MODEL_FORMAT_INI);
+    model->compile();
+    model->initialize();
   } catch (...) {
     std::cerr << "Error during init" << std::endl;
     return 0;
   }
 
   if (training) {
-    NN.setDataset(ml::train::DatasetModeType::MODE_TRAIN,
-                  std::move(data_train));
+    auto data_train = ml::train::createDataset(
+      ml::train::DatasetType::GENERATOR, getSample_train);
+
+    model->setDataset(ml::train::DatasetModeType::MODE_TRAIN,
+                      std::move(data_train));
 
     try {
-      NN.train({"save_path=" + weight_path});
+      model->train({"save_path=" + weight_path});
     } catch (...) {
       std::cerr << "Error during train" << std::endl;
       return 0;
     }
   } else {
     try {
-      NN.load(weight_path, ml::train::ModelFormat::MODEL_FORMAT_BIN);
+      model->load(weight_path);
     } catch (std::exception &e) {
       std::cerr << "Error during loading weights: " << e.what() << "\n";
       return 1;
     }
     std::ifstream dataFile(data_file);
     int cn = 0;
+    std::vector<float *> in;
+    std::vector<float *> l;
+
+    auto step = [](const float result) {
```

Comment:
Now we can delete stepFunction in line 58

Reply:
Oh. I didn't aware it is there. removed.